src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+                         int(self.parent._EBeamScanner.resolution.value[1] / self._shape.value[1]))
+
+        self._megafield_metadata = \
+            MegaFieldMetaData(
+                    mega_field_id=self.filename.value,
+                    pixel_size=self.parent._EBeamScanner.pixelSize.value[0],
+                    #TODO K.K. check input units vs units of VA's is it right that API takes seconds for dwell time?
+                    dwell_time=int(self.parent._EBeamScanner.dwellTime.value /
+                                   self.parent._EBeamScanner.clockPeriod.value),
+                    x_cell_size=self.cellCompleteResolution.value[0],
+                    x_eff_cell_size=eff_cell_size[0],
+                    y_cell_size=self.cellCompleteResolution.value[1],
+                    y_eff_cell_size=eff_cell_size[1],
+                    cell_parameters=[CellAcqParameters(translation[0], translation[1], darkOffset, digitalGain)
+                                     for translation, darkOffset, digitalGain in zip(cellTranslation,celldarkOffset,celldigitalGain)],
+                    x_scan_to_acq_delay=int(self.acqDelay.value - self.parent._EBeamScanner.scanDelay.value[0]),
+                    x_scan_delay=self.parent._EBeamScanner.scanDelay.value[0],
+                    y_prescan_lines=self.parent._EBeamScanner.scanDelay.value[1],
+                    flyback_time=int(self.parent._EBeamScanner.scanFlyback.value /
+                                   self.parent._EBeamScanner.clockPeriod.value),
+                    x_scan_offset=self.parent._EBeamScanner.scanOffset.value[0],
+                    y_scan_offset=self.parent._EBeamScanner.scanOffset.value[1],
+                    x_scan_gain=self.parent._EBeamScanner.scanGain.value[0],
+                    y_scan_gain=self.parent._EBeamScanner.scanGain.value[1],
+                    x_descan_gain=self.parent._MirrorDescanner.scanGain.value[0],
+                    y_descan_gain=self.parent._MirrorDescanner.scanGain.value[1],
+                    x_descan_offset=self.parent._MirrorDescanner.scanOffset.value[0],
+                    y_descan_offset=self.parent._MirrorDescanner.scanOffset.value[1],
+                    scan_rotation=self.parent._EBeamScanner.rotation.value,
+                    descan_rotation=self.parent._MirrorDescanner.rotation.value,
+            )
+
+        return self._megafield_metadata
+
+    def _acquire(self):
+        try:
+            self._current_status = None
+            tnext = 0
+
+            while True:
+                # Wait until a message is available
+                command, *meta_info = self.acq_queue.get(block=True)
+                if command == "start" and self._current_status != "start":
+                    self._current_status = command
+                    self.parent.simpleASMCall(self._server_URL + "/scan/start_mega_field", _METHOD_POST, 204, meta_info[
```

Comment:
I chose the current implementation to keep all the API calls packed together in the acuire method. I think that splitting hem up would result in less overview of where each call is made. Especially since the start, next, stop calls for interaction are already seperate methods and adding methods with more or less the same name and goal but different part in the process would be confusing. (There will be a start method putting stuff on the queue and one executing stuff from the queue, a next method putting stuff on the queue and one executing stuff from the queue etc.)
But maybe I miss something can you explain a bit more what you think would be good?